scripts/sign_encrypt.py

```diff
@@ -173,6 +175,17 @@ def main():
     with open(args.inf, 'rb') as f:
         img = f.read()
 
+    magic, img_type, img_size, alg, digest_len, sig_len = struct.unpack(
+        "IIIIHH", img[0:20])
+    if magic == 0x4f545348:
+        # Already encrypted. Cannot be sign-enc again
+        # Already signed. Skip header etc. to be able to re-sign
+        if img_type == 2:
+            logger.error('Sign-enc failed, the encrypted TA cannot be encrypted again.')
+            sys.exit(1)
+        # 20-header len,16-uuid len,4-version len
+        img = img[20 + 16 + 4 + sig_len + digest_len:]
```

Comment:
Verifying the signature might not be feasible at at this stage, but it should be possible to check the digest at least.
This should protect against unknowingly signing slightly broken or corrupted TAs.

Reply:
ok